### ISSUE TITLE
Host label affinity scheduling rule is not functional.

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#6163]
Rancher server version - v1.2.0-pre4-rc1

Steps to reproduce the problem:
Add a label (hello1=value1) to one of the hosts.

Add a service with scheduling rule - "host must have a host label hello1=value1"

service is stuck in "activating" state for ever and has error -

```(Failed to allocate instance [container:1i55]: Bad instance [container:1i55] in state [error]: Scheduling failed: needs host with label hello1: value1)```

Host has labels set:
 ```"labels": {
"io.rancher.host.agent_image": "rancher/agent:v1.1.0-rc1",
"io.rancher.host.docker_version": "1.12",
"io.rancher.host.linux_kernel_version": "4.4",
"hello1": "value1"
},```

On the service  following scehduling labels is set-

```"labels": {
"hello1": "value1",
"io.rancher.scheduler.affinity:host_label": "hello1=value1"
},```

**Comment by Dids** [comment]
I'm also seeing this, but only when specifying more than one label.

**Comment by tfiduccia** [comment]
I see this in pre4 and only with 1 label.

**Comment by cjellick** [comment]
Didnt do an explicit code fix, but the tidal wave of changes over the last few days appear to  have fixed this on master.

**Comment by cjellick** [comment]
@tfiduccia I'm assigning over to @sangeethah for verification but if you want to verify on master that your steps to repro are fixed too, go for it.

**Comment by sangeethah** [comment]
This issue still repros with latest master build.

**Comment by cjellick** [comment]
We narrowed it down to only occurring when there is an active docker machine driver. It is caused by the logic we have to combine machine and host into a single resource. @ibuildthecloud will address. Assigning over.

**Comment by sangeethah** [comment]
Tested with latest build form master.

Host label affinity scheduling rule works as expected.

**Comment by stobias123** [comment]
Same issue rancher v1.4.1. Seeing the below error. How can I troubleshoot this?
```
(Failed to allocate instance [container:1i97697]: Bad instance [container:1i97697] in state [error]: Allocation failed: needs host with label az: us-west-2a) 
```
Docker compose has 
```
      io.rancher.scheduler.affinity:host_label: az=us-west-2a
```
Rancher hosts shows
```

1j33     instance.internal    active    24           <IP ADDR>    az=us-west-2a,instance_id=i-ID,instance_type=r4.large,nginx_lb=true,traefik_lb=true
```

**Comment by bsidermob** [comment]
Not resolved in Rancher 1.6. The functionality doesn't work.

**Comment by superseb** [comment]
@stobias123 @bsidermob Can you check if you are hitting https://github.com/rancher/rancher/issues/10271. If not, please file a new issue with all information and a reproducible situation so we can take a look.
